src/api/v4/source/dataset.js

```diff
@@ -33,14 +33,18 @@ Dataset.prototype = Object.create(Base.prototype);
  * @param {Engine} engine - The engine object to be assigned to the internalModel.
  */
 Dataset.prototype._createInternalModel = function (engine) {
-  return new AnalysisModel({
+  var internalModel = new AnalysisModel({
     id: this.getId(),
     type: 'source',
     query: 'SELECT * from ' + this._dataset
   }, {
     camshaftReference: CamshaftReference,
     engine: engine
   });
+
+  internalModel.on('change:error', this._triggerError, this);
```

Comment:
Why not move this up to the base class so that we dont have to repeate it in `src/api/v4/source/sql.js`?